core/modules/date/date.field.inc

```diff
@@ -557,3 +677,442 @@ function date_field_settings_form($field, $instance, $has_data) {
   module_load_include('inc', 'date', 'date.admin');
   return _date_field_settings_form($field, $instance, $has_data);
 }
+
+/**
+ * Custom callback to provide the widget settings form.
+ *
+ * @param array $instance
+ *   Field instance definition.
+ *
+ * @return array
+ *   Widget settings form structure.
+ */
+function _date_html5_field_widget_settings_form(array $instance) {
+  $settings = $instance['widget']['settings'];
+  $form = array();
+
+  $form['limits'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Date input limits'),
+    '#description' => t('Applies to start and end if enabled and is relative to the current date.'),
```

Comment:
This will put it in line with the other date widgets.

```suggestion
    '#description' => t('Applies to start and end if enabled and is relative to the current date.'),
    '#fieldset' => 'date_format',
```

I agree they could all use improvement but better to have them in sync and fix them together.

Reply:
The `#fieldset` part currently doesn't do anything, as this only has a meaning in views - but we didn't add this widget to views exposed filters yet.

Reply:
Oh, not to forget - the `date_format` fieldset is the wrong one, as this is for "input formats" - which is something this widget doesn't use at all (the native html5 input elements have their own fixed input format).

Reply:
That is not correct; it's not just for Views. By adding `#fieldset` it rearranges the elements in `admin/structure/types/manage/*/fields/field_*`. I found out that Date module places the form elements in one of two fieldsets: either `date_format` or `default_values`. It uses an undocumented `#fieldset` form attribute (from what I can tell) that places the elements.

Then in `date.module` it reorganizes the settings for better presentation:

```
// Reorganize the instance settings and widget settings sections into a more
  // intuitive combined fieldset.
  $form['instance']['date_format'] = array(
    '#type' => 'fieldset',
    '#title' => t('Date entry'),
    '#collapsible' => TRUE,
    '#collapsed' => FALSE,
  );
  $form['instance']['default_values'] = array(
    '#type' => 'fieldset',
    '#title' => t('Default values'),
    '#collapsible' => TRUE,
    '#collapsed' => FALSE,
  );
```
In the old widgets the start/end year limits are actually part of `date_format` (aka Date entry) so to me it makes sense to put it there:

![2022-08-28 12 15 33 bd1 lndo site 7b490e6c2883](https://user-images.githubusercontent.com/1071483/187083962-2e1e1d67-6748-418e-9fc8-d54107c422ca.png)

When I add `'#fieldset' => 'date_format',` to the `limits` fieldset and `no_fieldset` checkbox it looks better and more organized like the old widgets:

![2022-08-28 12 15 06 bd1 lndo site f38a88fa0d21](https://user-images.githubusercontent.com/1071483/187083960-c5c389ff-a367-466c-8413-2ce718ad81c6.png)

I missed the `no_fieldset` checkbox in my previous suggestion. So I'm going to suggest these two changes again. I think it still makes more sense.

Date module is a bit odd, but at least if we follow the same pattern it'll be easier for someone to reorganize later.

Reply:
> It uses an undocumented #fieldset form attribute...

It's not that it's undocumented, one can just add any `#whatever` property to forms to re-use in later functions.

> Date module is a bit odd, but at least if we follow the same pattern

Odd? Insane! :wink:

I don't recommend to follow the same pattern. We provide the complete form, so we can rearrange however we like in function date_field_widget_form. The whole stuff in date_form_field_ui_field_edit_form_alter() and date_field_ui_field_edit_form_pre_render() is IMO unnecessary for our new widget, as we don't bloat the form that much.

There are items we simply do not need for this widget. Even if we followed the same pattern - we'd end up with a single checkbox in a fieldset. What for? (We don't need all that indirection.)

Here's a date field widget settings form to clarify what I mean:

![insane-date-widget](https://user-images.githubusercontent.com/842066/187144340-e62e5439-6955-430e-ae59-f371a5497cf2.png)